### DISCUSSION TITLE
building-asp-net-core-apps-using-cake-in-docker/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#54] [Announcements]
# building-asp-net-core-apps-using-cake-in-docker/

In this post, I show how to build an ASP.NET Core app by using Cake to build your project inside a Docker container, while taking advantage of layer caching

https://andrewlock.net/building-asp-net-core-apps-using-cake-in-docker/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 21 2017, 11:25, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/54#discussioncomment-6043159)</em>

---

<p>There actually _is_ globbing support, but it just doesn't work like you might expect unfortunately. It's basically expanded at the command line, which means you can copy multiple files, but they'll all end up in the same folder, instead of preserving the tree.</p><p>I expect you're right on the NET + .NET Core + .NET Mono image. From my point of view, this does everything I need, so I'm not too worried about an "official" image.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 15 2018, 10:20, in reply to [connorads](https://disqus.com/by/connorads/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/54#discussioncomment-6043163)</em>

---

<p>Hey, currently I do still use Mono, but that's something I'm hoping to review soon :) I don't Visual Studio Container Tools, so I haven't used that technique, but the approach described in that article should likely work without any issues.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 09 2018, 08:18, in reply to [Francisco Berrocal](https://disqus.com/by/franciscoberrocal/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/54#discussioncomment-6043165)</em>

---

<p>No problem, it's a good question :)</p><p>The main benefit is that you can easily do anything in the C# cake script, in a language that is familiar to you. Take the RunTests example - looping through each project in the tests directory - doing that with bash is not trivial to a newcomer. I'd wager that for many (most) C# developers, the default bash scripting used in most Dockerfiles would be an alien and frustrating thing to do complex tasks with.</p><p>In short, if it adds value, use it. There's obviously overhead associated with it, but that overhead is only at build-time, so if you find it adds value, go for it :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 05 2018, 03:02, in reply to [gakera](https://disqus.com/by/thingiverse-91e8b76fc5e5be729581044feca1a410/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/54#discussioncomment-6043168)</em>

---

<p>Ah, yeah that could be the case, I haven't tried a self-contained publish. If you remove the --no-restore, but still keep the restore stage, then you should still see some benefit from the layer caching. you may find the restore in the build is effectively a noop, or at the very least should reduce the amount it needs to pull from the network. I'm not 100% on that though, might be worth experimenting</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 06 2018, 08:33, in reply to [gakera](https://disqus.com/by/thingiverse-91e8b76fc5e5be729581044feca1a410/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/54#discussioncomment-6043171)</em>

---

<p>As far as I know, the Core CLR version of Cake is working well for people (there's even a .NET Core global tool for it). I haven't looked at this again yet, as the mono version has been working fine for me, and I'm using the reference assemblies at the moment. Once that dependency has gone, I'll probably take another look at the .NET Core version</p>

[a comment by an automated account is not shown]
